app/src/main/java/info/nightscout/androidaps/dialogs/CarbsDialog.kt

```diff
@@ -224,7 +225,7 @@ class CarbsDialog : DialogFragmentWithDate() {
                         }
 
                         eatingSoonSelected -> {
-                            uel.log("TT EATING SOON", d1 = eatingSoonTT, i1 = eatingSoonTTDuration)
+                            uel.log(Action.TT_EATING_SOON, ValueWithUnit(eatingSoonTT, if (units == Constants.MMOL) Units.Mmol_L else Units.Mg_Dl) , ValueWithUnit(eatingSoonTTDuration, Units.M))
```

Comment:
yes, exactly
create another constructor to bypass `if (units == Constants.MMOL) Units.Mmol_L else Units.Mg_Dl` for MMOL/MGDL units. Pass only `units`

Reply:
Constants class is not available in database module
I propose two solutions using a new constructor...

1) test in constructor ("mmol" hardcoded in it)
```
    data class ValueWithUnit (val dValue: Double, val iValue: Int, val lValue: Long, val unit: Units) {
        constructor(dvalue:Double, unit:Units) : this(dvalue, 0, 0, unit)
        constructor(ivalue:Int, unit:Units) : this(0.0, ivalue, 0, unit)
        constructor(lvalue:Long, unit:Units) : this(0.0, 0, lvalue, unit)
        constructor(dvalue:Double, bgunit:String) : this(dvalue, 0, 0, if (bgunit=="mmol") Units.Mmol_L else Units.Mg_Dl)
    }
```

2) move MGDL and MMOL definition in Units class
```
        @SerializedName("mg/dl") Mg_Dl,
        @SerializedName("mmol") Mmol_L,
```
in Constants class
```
public class Constants {
    public static final String MGDL = Units.Mg_Dl.name();        // This is Nightscout's representation
    public static final String MMOL = Units.Mmol_L.name();
```

```
constructor(dvalue:Double, unit:String) : this(dvalue, 0, 0, Units.fromString(unit))
```
I prefer second one...

Reply:
agree